.github/workflows/release_java.yml

```diff
@@ -0,0 +1,137 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Bindings Java Release
+
+on:
+  push:
+    tags:
+      # Staging JARs on Apache Nexus repository for RCs. Read more on
+      # https://opendal.apache.org/docs/contributing/release#release-maven-artifacts
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
+  workflow_dispatch:
```

Comment:
I have a few thoughts on these triggers:

1. I keep the `workflow_dispatch` so that we can somehow debug the workflow. It's possible that we retain only this trigger and let the RM trigger release staging on a tag manually. But let's try to do it fully automatically now.
2. For the tag triggers:

I create a pattern that only staging for each RC. This is because we only staging the artifacts instead of fire the release. The staging artifacts can be found at https://repository.apache.org/#stagingRepositories. For example, the last commit debugging commit staging `orgapacheopendal-1012`:

<img width="1531" alt="image" src="https://github.com/apache/incubator-opendal/assets/18818196/0c17e824-753a-4c9a-a23e-ea4c46ca456a">

I'll click "Drop" to clean up. And a RM should "Drop" on vote failed; "Close" and later "Release" on vote passed.

Note that since we tag `-rcN` Git tags now, other bindings releases should filter out the trigger on RCs, or following the process here to staging on RCs.